ccm_web/client/src/api.ts

```diff
@@ -1,36 +1,63 @@
+import { CanvasCourseBase } from './models/canvas'
 import { Globals, HelloMessageData } from './models/models'
 import handleErrors from './utils/handleErrors'
 
 export interface LtiProps {
   ltiKey: string | undefined
 }
 
-const createAuthHeaders = (key: string | undefined): RequestInit => {
+const initRequest = (key: string | undefined, headers: string[][] = []): RequestInit => {
   if (key !== undefined) {
+    headers.push(['Authorization', 'Bearer ' + key])
     return {
       credentials: 'include',
-      headers: { Authorization: 'Bearer ' + key }
+      headers: headers
     }
   }
   return {}
 }
 
+const getGet = (key: string | undefined): RequestInit => {
```

Comment:
Nit-pick: `getGet` and `getPut` seem like they are missing a word. I think we're trying to avoid `Headers` because there is a `headers` property in this object we're creating. Maybe `getGetRequestConfig`? I dunno...